packages/graphql-language-service-server/src/__tests__/MessageProcessor-test.js

```diff
@@ -0,0 +1,156 @@
+/**
+ *  Copyright (c) Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ *  @flow
+ */
+
+import {expect} from 'chai';
+import {Position} from 'graphql-language-service-utils';
+import {beforeEach, describe, it} from 'mocha';
+
+import * as handlers from '../MessageProcessor';
+import MockWatchmanClient from '../__mocks__/MockWatchmanClient';
+
+describe('MessageProcessor', () => {
+  const queryDir = `${__dirname}/__queries__`;
+
+  const textDocumentTestString = `
+  {
+    hero(episode: NEWHOPE){
+    }
+  }
+  `;
+
+  const initialDocument = {
+    textDocument: {
+      text: textDocumentTestString,
+      uri: `${queryDir}/test.graphql`,
+      version: 0,
+    },
+  };
+
+  async function setup() {
+    const params = {rootPath: __dirname};
+    const watchmanClient = new MockWatchmanClient();
+    const {capabilities} = await handlers.handleInitializeRequest(
+      params,
+      null,
+      null,
+      watchmanClient,
```

Comment:
I'm not really a fan of passing in `watchmanClient`, but for this PR this should be good. I'll have to modify some code so that this file is testable without looking at watchman instance.